RelationalAI/Errors/ApiException.cs

```diff
@@ -0,0 +1,66 @@
+﻿/*
+ * Copyright 2022 RelationalAI, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Net;
+using System.Net.Http.Headers;
+
+namespace RelationalAI.Errors
+{
+    /// <summary>
+    /// Represents error thrown when RAI API request failed with error status code, etc.
+    /// </summary>
+    public class ApiException : Exception
+    {
+        public ApiException(
+            string message,
+            HttpStatusCode statusCode,
+            string response,
+            HttpResponseHeaders headers)
+            : base($"Request failed with error: {message}")
+        {
+            StatusCode = statusCode;
+            Response = response;
+            Headers = headers;
```

Comment:
am curious why the api exception carries the HTTP headers? in the other SDKs we generally try to abstract away transport details at the API layer.

Reply:
@bradlo we really need that for the request id, as that's always important for tracing. We could extract the X-Request-ID header value and pass it directly if we know the exact header name.